Observer/SalesModelServiceQuoteSubmitBefore.php

```diff
@@ -42,8 +42,7 @@ public function execute(Observer $observer)
     {
         /** @var OrderInterface $order */
         $order = $observer->getEvent()->getOrder();
-
-        $quote = $this->quoteRepository->get($order->getQuoteId());
+        $quote = $observer->getEvent()->getQuote();
```

Comment:
@paveq , are you sure this is correctly working?
We intentionally loaded it from DB because it was not correctly working on our old Magento 2.0.7 installation, we had an empty quote object. Maybe a fixed bug?